mantidimaging/gui/windows/main/view.py

```diff
@@ -414,6 +414,11 @@ def closeEvent(self, event):
             # Ignore the close event, keeping window open
             event.ignore()
 
+    def cleanup(self):
+        # Release shared memory from loaded stacks
+        for stack in self.get_all_stacks():
```

Comment:
So this doesn't make any difference in terms of the memory being removed, however what seems to happen if it's not there is that we get resource tracker warnings when the user closes MI with data still loaded. It's as if the destructor that frees the memory is running after the resource tracker checks the status of things. I thought it might be worth having this in to get rid of the warnings, as they give the impression that MI hasn't ended cleanly. I also thought if we're seeing the warnings all the time then we might get into the habit of ignoring them and miss a genuine leak.